design/src/aws_config.md

```diff
@@ -90,3 +105,54 @@ Tokio/Hyper features.
 Currently, the `.build()` method on service config will fill in defaults. As part of this change, `.build()` called on
 the service config with missing properties would be a runtime panic. Most customers would utilize the `From`
 implementation.
+
+## Stability and Versioning
+
+The introduction of `SharedConfig` to aws-types is not without risks. If a customer depends on a version aws-config that
+uses `SharedConfig` that is incompatible, they will get confusing compiler errors.
+
+An example of a problematic version set:
+
+```
+┌─────────────────┐                 ┌───────────────┐
+│ aws-types = 0.1 │                 │aws-types= 0.2 │
+└─────────────────┘                 └───────────────┘
+           ▲                                 ▲
+           │                                 │
+           │                                 │
+           │                                 │
+ ┌─────────┴─────────────┐          ┌────────┴───────┐
+ │aws-sdk-dynamodb = 0.5 │          │aws-config = 0.6│
+ └───────────┬───────────┘          └───────┬────────┘
+             │                              │
+             │                              │
+             │                              │
+             │                              │
+             │                              │
+             ├─────────────────────┬────────┘
+             │ my-lambda-function  │
+             └─────────────────────┘
+```
+
+To mitigate this risk, we will instead introduce `aws-types`. `aws-core` is intended to be permanently stable. Changes
+to `aws-types` need to be made with extreme care. Likely, `aws-core` will launch at `0.1` and only experience minor
+version bumps until a single breaking change at 1.0.
```

Comment:
I'm confused on this. `aws-types` already exists. Do you mean introduce `aws-core`? Would also be helpful to have a diagram of this like the one above showing the problem addressed.